lib/testutils/httpmultibin/httpmultibin.go

```diff
@@ -108,7 +108,14 @@ type jsonBody struct {
 
 func getWebsocketHandler(echo bool, closePrematurely bool) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		conn, err := (&websocket.Upgrader{}).Upgrade(w, req, w.Header())
+
+		// Echo back User-Agent header if it exists
+		responseHeaders := w.Header().Clone()
+		if ua := req.Header.Get("User-Agent"); ua != "" {
+			responseHeaders.Add("Echo-User-Agent", req.Header.Get("User-Agent"))
+		}
+
+		conn, err := (&websocket.Upgrader{}).Upgrade(w, req, responseHeaders)
```

Comment:
Can you please instead of changing this make a custom handler just for you test and add it just for your test as it's done in https://github.com/grafana/k6/blob/f76bfa5a34ba9af7e0165cb7016167ffed7dcb91/js/modules/k6/http/tls_test.go#L40-L43 .

You can skip most of the rest of the code after the `Upgrade` ... I think :thinking:

Reply:
thanks for the code example.. !